### PR TITLE
Update kotlin-dsl plugin to 4.0.11

### PR DESCRIPTION
**Pull request by asodja**
For Kotlin 1.8.20, see https://github.com/gradle/gradle/pull/24626.

### CODE REVIEW

File: subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/AbstractInitIntegrationSpec.groovy [hunk 1]
```diff
@@ -37,6 +37,7 @@ abstract class AbstractInitIntegrationSpec extends AbstractIntegrationSpec {
         file("settings.gradle") << """
             // This is here to prevent Gradle searching up to find the build's settings.gradle
         """
+        file("gradle.properties") << "\nkotlin.incremental.useClasspathSnapshot=false\n"
```
Comment (eskatos) [inline subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/AbstractInitIntegrationSpec.groovy:40]: A comment should link to the root cause issue at https://youtrack.jetbrains.com/issue/KT-57757/Reduce-classpath-snapshotter-memory-consumption

File: subprojects/samples/src/integTest/groovy/org/gradle/integtests/samples/java/SamplesJavaTestingIntegrationTest.groovy [hunk 2]
```diff
@@ -106,6 +106,8 @@ class SamplesJavaTestingIntegrationTest extends AbstractSampleIntegrationTest {
         given:
         TestFile dslDir = sample.dir.file(dsl)
         executer.inDirectory(dslDir)
+        // To workaround failure with Kotlin 1.8.20
```
Comment (eskatos) [inline subprojects/samples/src/integTest/groovy/org/gradle/integtests/samples/java/SamplesJavaTestingIntegrationTest.groovy:109]: A comment should link to the root cause issue at https://youtrack.jetbrains.com/issue/KT-57757/Reduce-classpath-snapshotter-memory-consumption

**Review by eskatos — APPROVED** [review]

**Comment by asodja** [comment]
@bot-gradle test QF

File: build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts [hunk 3]
```diff
@@ -224,6 +224,10 @@ fun Test.configureJvmForTest() {
             jvmArgs(listOf("--add-opens", "java.base/java.lang=ALL-UNNAMED")) // Used in tests by ClassLoaderUtils
         }
     }
+    if (usesEmbeddedExecuter() && OperatingSystem.current().isWindows) {
+        // Disable incremental snapshotting for Windows, since it throws OOM
```
Comment (eskatos) [inline build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts:228]: The comment should link to the root cause issue at https://youtrack.jetbrains.com/issue/KT-57757/Reduce-classpath-snapshotter-memory-consumption

File: subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/AbstractInitIntegrationSpec.groovy (hunk 1 above)
Comment (eskatos) [reply subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/AbstractInitIntegrationSpec.groovy:40]: This should be removed now that we do the same for all windows embedded tests

File: subprojects/samples/src/integTest/groovy/org/gradle/integtests/samples/java/SamplesJavaTestingIntegrationTest.groovy (hunk 2 above)
Comment (eskatos) [reply subprojects/samples/src/integTest/groovy/org/gradle/integtests/samples/java/SamplesJavaTestingIntegrationTest.groovy:109]: This should be removed now that we do the same for all windows embedded tests

File: subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/AbstractInitIntegrationSpec.groovy (hunk 1 above)
Comment (asodja) [reply subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/AbstractInitIntegrationSpec.groovy]: Done, dropped the commit.

File: subprojects/samples/src/integTest/groovy/org/gradle/integtests/samples/java/SamplesJavaTestingIntegrationTest.groovy (hunk 2 above)
Comment (asodja) [reply subprojects/samples/src/integTest/groovy/org/gradle/integtests/samples/java/SamplesJavaTestingIntegrationTest.groovy]: Done, dropped the commit.

**Comment by asodja** [comment]
@bot-gradle test and merge

**Review by eskatos — APPROVED** [review]

**Pull request merged**